### PR TITLE
feat: add instrumentation scope support

### PR DESCRIPTION
**Pull request by nityanandagohain**
Otel collector changes for  https://github.com/SigNoz/signoz/issues/2705

### CODE REVIEW

**Comment by srikanthccv** [comment]
It also has attributes https://github.com/open-telemetry/opentelemetry-proto/blob/bd7cf55b6d45f3c587d2131b68a7e5a501bdb10c/opentelemetry/proto/common/v1/common.proto#L69-L81; Do we not want to support them for now?

**Comment by nityanandagohain** [comment]
I was planning to just introduce the name first, but I guess it's better to introduce the version as well.

**Comment by srikanthccv** [comment]
I suggest adding the attributes as well so you don't have to come back to it again in future if somebody asks for it.

**Comment by nityanandagohain** [comment]
hmm makes sense, regarding the implementation of it do see any cons with the attributes in a similar way to that of resource attributes. Here will we will name it as instrumentation_scope_string_key, instrumentation_scope_string_value.

**Comment by srikanthccv** [comment]
I don't see any issue.

**Comment by nityanandagohain** [comment]
have added support for name, version and attributes.

**Review by srikanthccv — APPROVED** [review]

File: exporter/clickhouselogsexporter/exporter.go [hunk 1]
```diff
@@ -249,6 +249,17 @@ func (e *clickhouseLogsExporter) pushToClickhouse(ctx context.Context, ld plog.L
 			resources = addTemporaryUnderscoreSupport(resources)
 
 			for j := 0; j < logs.ScopeLogs().Len(); j++ {
+				scope := logs.ScopeLogs().At(j).Scope()
+				scopeName := scope.Name()
+				scopeVersion := scope.Version()
+
+				scopeAttributes := attributesToSlice(scope.Attributes(), true)
+
+				err := addAttrsToTagStatement(tagStatement, "instrumentation_scope", resources)
```
Comment (srikanthccv) [inline exporter/clickhouselogsexporter/exporter.go:258]: This should be using `scopeAttributes`

Comment (nityanandagohain) [reply exporter/clickhouselogsexporter/exporter.go:258]: Have updated it to `scope` https://github.com/SigNoz/signoz-otel-collector/pull/344

**Pull request merged**